dbt/adapters/redshift/connections.py

```diff
@@ -240,8 +240,9 @@ class RedshiftConnectionManager(SQLConnectionManager):
     def _get_backend_pid(self):
         sql = "select pg_backend_pid()"
         _, cursor = self.add_query(sql)
+
         res = cursor.fetchone()
-        return res
```

Comment:
I think it's safe to assume that we can always get a PID. If we can't, there are probably worse things going on. And to clarify a comment in the PR description, I think what happened is that moving from `psycopg2` to `redshift-connector` changed the behavior in this way:

```
> cursor.fetchone()
> <my_pid>
```
```
> cursor.fetchone()
> (<my_pid>,)
```

It's worth noting that the latter is [PEP-249](https://peps.python.org/pep-0249/#fetchone) compliant. This would align with other issues we've seen where `psycopg2` is not PEP-249 compliant but `redshift-connector` is.

tl;dr; The way you have it is fine, I'm just providing context.

Reply:
🧠 Thanks for this context